code/parse/sexp.cpp

```diff
@@ -20298,6 +20300,49 @@ void sexp_string_concatenate(int n)
 }
 
 // Goober5000
+void sexp_string_concatenate_block(int n)
+{
+	int sexp_variable_index;
+	SCP_string new_text;
+
+	// Only do single player or multi host
+	if (MULTIPLAYER_CLIENT)
+		return;
+
+	// get sexp_variable index
+	Assert(Sexp_nodes[n].first == -1);
+	sexp_variable_index = atoi(Sexp_nodes[n].text);
+	n = CDR(n);
+
+	// verify variable set
+	Assert(Sexp_variables[sexp_variable_index].type & SEXP_VARIABLE_SET);
+
+	// check variable type
+	if (!(Sexp_variables[sexp_variable_index].type & SEXP_VARIABLE_STRING))
+	{
+		Warning(LOCATION, "Cannot assign a string to a non-string variable!");
+		return;
+	}
+
+	// concatenate strings
+	while (n >= 0)
+	{
+		new_text.append(CTEXT(n));
+		n = CDR(n);
+	}
+
+	// check length
+	if (new_text.length() >= TOKEN_LENGTH)
+	{
+		Warning(LOCATION, "Concatenated string '%s' is too long and will be truncated.", new_text.c_str());
```

Comment:
This warning could include the maximum size of the string. Otherwise the user would need to manually count the characters in the truncated string.

The documentation of this SEXP already contains this information but it would also be useful to see that here.